### PR TITLE
Comment PR

### PR DESCRIPTION
**Pull request by dizko**
Just creating this PR to add some comments. DO NOT MERGE.

### CODE REVIEW

**Comment by dizko** [comment]
I would start by keeping your branches smaller in scope, this one has over 1000 additions. Try to think of branches as smaller feature sets. For example...
If I wanted to work on the feature where I can turn a knob with my mouse, when I create a branch in git, I would write `git branch feature/knob-turning-via-keyboard`. I would then focus on that feature, once complete I would put in a PR for that to merge into the `dev` branch.

I would make `dev` the project's primary branch. `master` should remain untouched until you're about to release, then you would merge `dev` into `master` and cut a tag off of master to make your release build off of. This is so that you snap shot the application at that moment in time. (with the tag)

It's also not a bad idea to categorize your branching, but this practice will vary from shop to shop, ex:
`feature/some-feature`
`defect/BUG-2341-thing-is-broken`

Also, comment on your commits on what you actually did, it doesn't matter for working on a project by yourself but it's a good practice to get into. Development isn't just writing the code, it's doing all this little stuff too.

This is a good explanation of a general git flow:
https://nvie.com/posts/a-successful-git-branching-model/

File: src/render.component.js [hunk 1]
```diff
@@ -28,10 +29,10 @@ class RenderComponent {
 
         //hang DOM elements off parent elements
         header.appendChild(main);
-        main.appendChild(synthControlDiv);
+        main.appendChild(this.synthControlDiv);
         main.appendChild(synthTriggerParentDiv);
-        synthControlDiv.appendChild(leftPanel);
-        synthControlDiv.appendChild(rightPanel);
+        this.synthControlDiv.appendChild(leftPanel);
+        this.synthControlDiv.appendChild(rightPanel);
```
Comment (dizko) [inline src/render.component.js]: This is hard to follow in my opinion. There isn't much benefit in creating the DOM through javascript in this manner, I don't see a reason why this couldn't simply be part of the static HTML in `index.html`. Or, at the very least write all this HTML out as a template string.
```
let template = `
    <header>
        <div class="synthControls"></div>
    </header>`
```

I also would keep your CSS classes like `synth-controls` vs camel case, I would reserve camel case for JS.

Then I would query the DOM when needed and storing the reference at that time.

Comment (dizko) [reply src/render.component.js]: To follow up, if you need a reference to `synthControlDiv` I would create a method on the class:
`get sythControlDiv() { return this.sythControlDiv = this.synthControlDiv ? document.querySelector('.synth-control') : this.synthControlDiv; });`

**Comment by kdfriedman** [comment]
I really appreciate the tips and advice via this repo, thanks so much for taking a look. Regarding the creation of DOM elements via JS, I wanted to try and mimic what a modern framework might leverage via HTML templates but could have definitely architected much differently. As you said, template strings would have been a much less verbose and cleaner choice in this instance. I'm also going to be much more conscientious via my Git branches and commits, your comments are very helpful and informative.

File: src/render.component.js (hunk 1 above)
Comment (kdfriedman) [reply src/render.component.js]: I really appreciate the tips and advice via this repo, thanks so much for taking a look. Regarding the creation of DOM elements via JS, I wanted to try and mimic what a modern framework might leverage via HTML templates but could have definitely architected much differently. As you said, template strings would have been a much less verbose and cleaner choice in this instance. I'm also going to be much more conscientious via my Git branches and commits, your comments are very helpful and informative.

Comment (kdfriedman) [reply src/render.component.js]: I really appreciate the tips and advice via this repo, thanks so much for taking a look. Regarding the creation of DOM elements via JS, I wanted to try and mimic what a modern framework might leverage via HTML templates but could have definitely architected much differently. 

As you said, template strings, 
`const HTMLBoilerplate = 
`<header class="header-1"></header>
     <main>
         <section class="synth-controls"></section>
     </main>``

would have been a much less verbose and cleaner choice in this instance. I'm also going to be much more conscientious via my Git branches and commits, your comments are very helpful and informative.
